toolchain/sem_ir/function.h

```diff
@@ -15,38 +15,42 @@ namespace Carbon::SemIR {
 struct Function : public Printable<Function> {
   // A value that describes whether the function uses a return slot.
   enum class ReturnSlot : int8_t {
-    // Not yet known: the function has not been called or defined.
-    NotComputed,
     // The function is known to not use a return slot.
     Absent,
     // The function has a return slot, and a call to the function is expected to
     // have an additional final argument corresponding to the return slot.
     Present,
-    // Computing whether the function should have a return slot failed, for
-    // example because the return type was incomplete.
-    Error
+    // Computing whether the function should have a return slot failed because
+    // the return type was incomplete.
+    Incomplete,
+  };
+
+  // Information about how a function returns its return value.
+  struct ReturnInfo {
```

Comment:
FWIW, had you considered:

SemIR::Function::ReturnInfo -> SemIR::ReturnSlot
SemIR::Function::ReturnSlot -> SemIR::ReturnSlot::Kind

The Kind enum probably won't be used much outside (I do see the switches), but maybe naming this `ReturnSlot` would be helpful? And while the nesting of `ReturnSlot` in `Function` made sense when it was for `Function::return_slot`, it seems odder to me when `Function::ReturnSlot` is for `ReturnInfo::slot` versus having it be `ReturnInfo::ReturnSlot` (or as in the example above, `ReturnSlot::Kind`).

Reply:
I think I can remove `ReturnSlot` entirely. Will investigate in a separate PR.

Reply:
For now I've moved these two types out of `Function`.